julia-paren-indent skips blanks after paren

dfannius opened the pull request:
Some coders prefer to put some space inside their parentheses in some circumstances, e.g.,

```
foo( bar,
     baz,
     quux )
```

rather than

```
foo(bar,
    baz,
    quux)
```

julia-mode has been indenting the first example as

```
foo( bar,
    baz,
    quux )
```

which is ugly.

This patch indents both the first and second examples nicely by looking for blank space after the open paren. As before, If a line ends with a paren, the next line will begin one column to the right of it.

Note that the original docstring was incorrect, as julia-paren-indent was always returning the column position _after_ the innermost containing paren before point. So this is a smaller change than the docstring change would indicate.

dfannius committed:
julia-paren-indent skips blanks after paren

Wilfred commented on contrib/julia-mode.el:
```diff
@@ -356,8 +356,9 @@ This variable has a moderate effect on indent performance if set too
-  "Return the column position of the innermost containing paren
-before point. Returns nil if we're not within nested parens."
+  "Return the column position of the first non-blank character
+after the innermost containing paren before point. Returns nil if
```
You're absolutely right that the previous docstring had an off-by-one error, however I find this longer sentence hard to mentally parse. How about "Return the column position required to line up with the containing paren."?

Wilfred commented:
Seems totally reasonable to me, but this is the sort of functionality that could easily get broken in a refactoring. Would you mind adding a test?

dfannius committed:
julia-paren-indent: docstring, test

dfannius commented:
Thanks for the comments. I added a test and updated the docstring. I thought it was important to note what it is actually doing so that readers can understand why the new code is in there, but I simplified the description and added a phrase about the intent. If anyone wants to rephrase, I'm not wedded to the wording.

pao commented:
@Wilfred is this just waiting on someone to merge?

Wilfred commented:
Yep, I don't have the power to merge. Feel free to do the honours :)

yuyichao commented:
Done. Thanks.
